src/amltk/scheduling/plugins/emissions_tracker_plugin.py

```diff
@@ -0,0 +1,118 @@
+from codecarbon import EmissionsTracker
+from typing import Callable, Any, ClassVar, TypeVar
+from typing_extensions import Self
+
+from amltk.scheduling.plugins.plugin import Plugin
+from amltk.scheduling.task import Task
+
+P = TypeVar("P")
+R = TypeVar("R")
+
+
+class _EmissionsTrackerWrapper:
+    """A wrapper around codecarbon package to measure emissions."""
+
+    def __init__(
+            self,
+            fn: Callable[[P], R],
+            task: Task,
+            *codecarbon_args: Any,
+            **codecarbon_kwargs: Any,
+    ):
+        """Initialize the wrapper.
+
+        Args:
+            fn: The function to wrap.
+            task: The task the function is being wrapped for.
+            *codecarbon_args: arguments to pass to
+                [`codecarbon.EmissionsTracker`][codecarbon.EmissionsTracker].
+            **codecarbon_kwargs: keyword arguments to pass to
+                [`codecarbon.EmissionsTracker`][codecarbon.EmissionsTracker].
```

Comment:
This likely won't work without having the "inventory" of their documentation. I'm not sure that have one but you can see which one's it's enabled for here in the [`mkdocs.yaml`](https://github.com/automl/amltk/blob/2960fc4303d08a5aacb7d7c4e3187f78ef6b7376/mkdocs.yml#L111-L125) file.

Might be better to just leave a link a raw html link to their documentation as we don't use their docs anywhere else really.